ksqldb-engine/src/main/java/io/confluent/ksql/ddl/commands/DropTypeFactory.java

```diff
@@ -15,14 +15,32 @@
 
 package io.confluent.ksql.ddl.commands;
 
+import com.google.common.annotations.VisibleForTesting;
 import io.confluent.ksql.execution.ddl.commands.DropTypeCommand;
+import io.confluent.ksql.metastore.MetaStore;
 import io.confluent.ksql.parser.DropType;
+import io.confluent.ksql.util.KsqlException;
+
+import java.util.Objects;
 
 public class DropTypeFactory {
-  DropTypeFactory() {
+  private final MetaStore metaStore;
+
+  @VisibleForTesting
+  DropTypeFactory(final MetaStore metaStore) {
+    this.metaStore = Objects.requireNonNull(metaStore, "metaStore");
   }
 
   public DropTypeCommand create(final DropType statement) {
-    return new DropTypeCommand(statement.getTypeName());
+    final String typeName = statement.getTypeName();
+    final boolean ifExists = statement.getIfExists();
+
+    if (!metaStore.resolveType(typeName).isPresent()) {
+      if (!ifExists) {
+        throw new KsqlException("Type " + typeName + " does not exist.");
+      }
+    }
```

Comment:
No need to search the metastore when `ifExists` is `true`.

```suggestion
    if (!ifExists && !metaStore.resolveType(typeName).isPresent()) {
      throw new KsqlException("Type " + typeName + " does not exist.");
    }
```

Reply:
Done